Verify links and anchors in the documentation

Vampire opened the pull request:
Fix dead links in docs

Verify no dead links are present in docs

Fix dead anchors in docs

Verify no dead anchors are present in docs

Unify manual anchor style in docs

Verify no duplicate anchors are present in docs

Vampire commented:
* **#1911** <a href="https://app.graphite.dev/github/pr/spockframework/spock/1911?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Vampire and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

leonard84 commented on build.gradle:
```diff
@@ -408,6 +417,87 @@ tasks.named("asciidoctor") {
     }
   }
+
+  doLast {
```
:x: We should not add any more build logic into this file. I've been debating whether the docs need their own dedicated project or not. 
In any case, this logic is complex abd should be moved to a plugin in the `build-logic` module and have proper tests of their own.

Vampire replied:
I moved the logic and added some tests

leonard84 commented on docs/data_driven_testing.adoc:
```diff
@@ -545,7 +545,7 @@ string, this value is used as unroll-pattern. This could for example be set to
   a data variable called `iterationName` that is then used for reporting
 
-[[unroll_tokens]]
+[[unroll-tokens]]
```
🤔 I'm not sure we should change existing section links, as that will break if you have an old link and just change the version number in the URL.

Vampire replied:
We never really cared about that.
Everytime you add an explicit id to a section you effectively change its ID in the produced docs, as we never chose the one that was generated in the one-page variant by default.

So I thought this is not really too much an issue for us and it would be better to change the foreigners to the consistent naming scheme?

leonard84 commented:
We should probably also add some `fatalWarnings` ([doc](https://asciidoctor.github.io/asciidoctor-gradle-plugin/master/user-guide/)), such as `missingIncludes()`.

I've also added a workflow to actually run the docs task during the PR.

Vampire commented:
> We should probably also add some fatalWarnings ([doc](https://asciidoctor.github.io/asciidoctor-gradle-plugin/master/user-guide/)), such as missingIncludes().

Ah, nice, added.

> I've also added a workflow to actually run the docs task during the PR.

I'm not sure whether you have seen it, but I already did that here by simply adding `asciidoctor` for `ghActionsBuild`.
Does that mean you want me to remove that again?

leonard84 commented:
> I'm not sure whether you have seen it, but I already did that here by simply adding asciidoctor for ghActionsBuild.
> Does that mean you want me to remove that again?

Yes, we only build the docs for a single combination on release https://github.com/spockframework/spock/blob/247031f5787199c14129fd72ad724ade953bc8e8/.github/workflows/release.yml#L92-L99.

Vampire commented:
Ok, removed
